src/Elastic.Markdown/Slices/DescriptionGenerator.cs

```diff
@@ -0,0 +1,132 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System.Text;
+using Elastic.Markdown.Myst.Directives;
+using Elastic.Markdown.Myst.Substitution;
+using Markdig.Syntax;
+using Markdig.Syntax.Inlines;
+
+namespace Elastic.Markdown.Slices;
+
+public interface IDescriptionGenerator
+{
+	string GenerateDescription(MarkdownDocument document);
+}
+
+
+public class DescriptionGenerator : IDescriptionGenerator
+{
+	private const int MaxLength = 150;
+
+	public string GenerateDescription(MarkdownDocument document)
+	{
+		var description = new StringBuilder();
+		foreach (var block in document.TakeWhile(_ => description.Length < MaxLength))
+		{
+			switch (block)
+			{
+				case IncludeBlock include:
+					{
+						ProcessIncludeBlock(include, description);
+						break;
+					}
+				case ParagraphBlock paragraph:
+					{
+						ProcessParagraph(paragraph, description);
+						break;
+					}
+				case ListBlock listBlock:
+					{
+						ProcessListBlock(listBlock, description);
+						break;
+					}
+			}
+		}
+
+		var result = description.ToString().TrimEnd('.');
+		// It can happen that the last parsed block is longer, hence the result is longer than maxLength
+		// Hence we need to shorten it. In this case it will be shorted to until the next space after `MaxLength`
+		if (result.Length > MaxLength)
+		{
+			var endIndex = result.IndexOf(' ', MaxLength - 1);
+			if (endIndex == -1)
+				endIndex = MaxLength;
+			result = string.Concat(result.AsSpan(0, endIndex + 1).Trim(), "...");
+		}
+
+		return result;
+	}
+
+#pragma warning disable IDE0060
+	private static void ProcessIncludeBlock(IncludeBlock include, StringBuilder description) => Console.WriteLine("Not implemented: ProcessIncludeBlock");
```

Comment:
```suggestion
	private static void ProcessIncludeBlock(IncludeBlock include, StringBuilder description) {}
```

Or use our logging infrastructure.

Reply:
You could use discard parameters to not have the `#pragma`

Reply:
Yeah. Interestingly, I had an error saying something like `Duplicate arg "_" something something`.

But I'm going to clean this up and just add a comment instead.

Reply:
https://github.com/elastic/docs-builder/pull/778/commits/1f619098daff40ac053c937bfc6d5e6fc0d78bae